### PR TITLE
adds framework for bounties unlocking new crates and adds a few as an alterantive to mining tech

### PR DESCRIPTION
**Pull request by SomeguyManperson**
# General Documentation
numbers not final, let me know if they're too cheap
### Intent of your Pull Request

adds progression bounties, which unlock the ability to order new crates on completion, and puts important mining gear behind them
since mining gear is also used by things without access to a cargo shuttle mining tech isn't yeeted 
yet

### Why is this change good for the game?
reduces the reliance of mining on science by allowing them to rely on a department that profits off their successes, and the requirement for science to burn points on a tech tax to not stall the round
# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
three new bounties that unlock the ability to purchase KA modkits, plasmacutters, and advanced plasmacutters respectively

# Changelog

:cl:  
rscadd: new cargo bounties and a system to allow bounties to unlock the ability to order new crates
tweak: miners can now do bounties to unlock mining gear so they can wait for the cargo shuttle instead of waiting for science
/:cl:

### CODE REVIEW

File: code/modules/cargo/bounties/progression.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	var/list/unlocked_crates = list()
+
+/datum/bounty/progression/reward_string()
+	return "[reward] Credits and clearance to order new crates"
+
+/datum/bounty/item/progression/claim(mob/user)
+	..()
+	for(var/i in unlocked_crates)
+		var/datum/supply_pack/to_unlock = SSshuttle.supply_packs[i]
+		to_unlock.special_enabled = TRUE
```
Comment (redmoogle) [inline code/modules/cargo/bounties/progression.dm:11]: unlocked may be a better name

**Comment by ToasterBiome** [comment]
:(

File: code/modules/cargo/bounties/progression.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/datum/bounty/item/progression/mining_basic
+	name = "Common Mineral Prospecting"
+	description = "Basic materials are worth pocket change, but are integral for station longevity. Ship us a sheet of gold, uranium, or silver to certify your mining program as \"functional\""
+	reward = 1000
+	wanted_types = list(/obj/item/stack/sheet/mineral/silver,/obj/item/stack/sheet/mineral/gold,/obj/item/stack/sheet/mineral/uranium)
+	unlocked_crates = list(/datum/supply_pack/clearance/ka_damage,/datum/supply_pack/clearance/ka_cooldown,/datum/supply_pack/clearance/ka_range)
+
+/datum/bounty/item/progression/mining_basic/reward_string()
```
Comment (redmoogle) [inline code/modules/cargo/bounties/progression.dm:20]: May be better to make this a var

File: code/modules/cargo/bounties/progression.dm (hunk 1 above)
Comment (SomeguyManperson) [reply code/modules/cargo/bounties/progression.dm:11]: special_enabled was used for other special crates already like station goals which is why it's used here

**Comment by LazennG** [comment]
fuck science

File: code/modules/cargo/bounties/progression.dm (hunk 1 above)
Comment (redmoogle) [reply code/modules/cargo/bounties/progression.dm:11]: Ah ok _still think its better_

**Comment by Mesalikepie** [comment]
I prefer it to stay in science but this is an alternative which is acceptable. Test merge perhaps?

File: code/modules/cargo/bounties/progression.dm (hunk 2 above)
Comment (SomeguyManperson) [reply code/modules/cargo/bounties/progression.dm:20]: out of scope for this pr B)
